src/pages/index.js

```diff
@@ -87,39 +167,48 @@ function getCardElement(data) {
   const cardNameElement = cardElement.querySelector(".card__title");
   const cardImage = cardElement.querySelector(".card__image");
   const cardLikeButton = cardElement.querySelector(".card__like-button");
-
   const cardDeleteButton = cardElement.querySelector(".card__delete-button");
 
+  // if the card is liked, set the active class on the card
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
   cardNameElement.textContent = data.name;
   cardImage.src = data.link;
   cardImage.alt = data.name;
 
-  cardLikeButton.addEventListener("click", () => {
-    cardLikeButton.classList.toggle("card__like-button_liked");
-  });
+  cardLikeButton.addEventListener("click", (event) =>
+    handleLike(event, data._id)
+  );
 
   cardDeleteButton.addEventListener("click", () => {
-    cardElement.remove();
+    handleDeleteCard(cardElement, data._id);
+    selectedCardId = data._id;
   });
 
-  cardImage.addEventListener("click", () => {
-    openModal(previewModal);
+  // cardImage.addEventListener("click", () => {
+  //   openModal(previewModal);
+
+  // cardImage.addEventListener("click", () => handleImageClick(data));
+
+  previewModalCaptionElement.textContent = data.name;
+  previewModalImageElement.src = data.link;
+  previewModalImageElement.alt = data.name;
```

Comment:
* The `image modal` doesn't open when I click on   cards

This should be inside the image click handler (not just in the function body)


```js
  cardImage.addEventListener("click", () => {
    previewModalCaptionElement.textContent = data.name;
    previewModalImageElement.src = data.link;
    previewModalImageElement.alt = data.name;
    openModal(previewModal);
  });
```